File: code-coaches/medium/no-numerals/no_numerals.cpp
#include <iostream>
#include <string>
#include <map>

std::string replace_numbers_with_words(std::string sentence)
{
    std::map<int, std::string> num_to_word = {
        {0, "zero"}, {1, "one"}, {2, "two"}, {3, "three"}, {4, "four"},
        {5, "five"}, {6, "six"}, {7, "seven"}, {8, "eight"}, {9, "nine"},
        {10, "ten"}
    };

    for (int i = 0; i < sentence.length(); i++)
    {
        if (std::isdigit(sentence[i]))
        {
            int num = sentence[i] - '0';
            i++;

            // Check if there are additional digits that form a multi-digit number
            while (i < sentence.length() && std::isdigit(sentence[i])) {
                num = num * 10 + (sentence[i] - '0');
                i++;
            }

            // Replace the number with its English word form if it's between 0 and 10
            if (num >= 0 && num <= 10)
            {
                sentence.replace(i - std::to_string(num).length(), std::to_string(num).length(), num_to_word[num]);
                i += num_to_word[num].length() - std::to_string(num).length();
            }
        }
    }

    return sentence;
}

int main()
{
    std::string sentence;
    std::getline(std::cin, sentence);

    std::cout << replace_numbers_with_words(sentence);

    return 0;
}

File: code-coaches/medium/no-numerals/no_numerals.rb
def replace_numbers_with_words(sentence)
    number_words = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine", "ten"]

    words = sentence.split
    updated_words = words.map do |word|
        if word.to_i.to_s == word && word.to_i <= 10
            number_words[word.to_i]
        else
            word
        end
    end

    updated_words.join(" ")
end


def main
    sentence = gets.chomp
    puts replace_numbers_with_words(sentence)
end


if __FILE__ == $0
    main
end